src/uct/ib/rc/accel/rc_mlx5_ep.c

```diff
@@ -185,30 +188,32 @@ ucs_status_t uct_rc_mlx5_ep_put_zcopy(uct_ep_h tl_ep, const uct_iov_t *iov, size
                        "uct_rc_mlx5_ep_put_zcopy");
     UCT_CHECK_LENGTH(uct_iov_total_length(iov, iovcnt), 0, UCT_IB_MAX_MESSAGE_SIZE,
                      "put_zcopy");
-    UCT_RC_CHECK_RES(&iface->super, &ep->super);
-
-    uct_rc_mlx5_ep_fence_put(iface, &ep->tx.wq, &rkey, &remote_addr,
-                             ep->super.atomic_mr_offset);
-
-    status = uct_rc_mlx5_ep_zcopy_post(ep, MLX5_OPCODE_RDMA_WRITE, iov, iovcnt,
-                                       0ul, 0, NULL, 0, remote_addr, rkey,
-                                       0ul, 0, 0, MLX5_WQE_CTRL_CQ_UPDATE,
-                                       uct_rc_ep_send_op_completion_handler, 0,
-                                       comp);
-    UCT_TL_EP_STAT_OP_IF_SUCCESS(status, &ep->super.super, PUT, ZCOPY,
+    UCT_RC_CHECK_RES(&iface->super, &ep->super.super);
+
+    uct_rc_mlx5_ep_fence_put(iface, &ep->super.tx.wq, &rkey, &remote_addr,
+                             ep->super.super.atomic_mr_offset);
+
+    status = uct_rc_mlx5_base_ep_zcopy_post(&ep->super, MLX5_OPCODE_RDMA_WRITE,
+                                            iov, iovcnt, 0ul, 0, NULL, 0,
+                                            remote_addr, rkey, 0ul, 0, 0,
+                                            MLX5_WQE_CTRL_CQ_UPDATE,
+                                            uct_rc_ep_send_op_completion_handler,
+                                            0, comp);
+    UCT_TL_EP_STAT_OP_IF_SUCCESS(status, &ep->super.super.super, PUT, ZCOPY,
```

Comment:
why nor declaring base ep instead?
`UCT_RC_MLX5_BASE_EP_DECL(tl_ep, iface, ep);`

Why is it handled differently in different functions?